### PR TITLE
validate typings

### PR DESCRIPTION
**Pull request by mgrybyk**
## Proposed changes

POC: typings validation

We had repetitive problems because of typos in jsdoc/etc that causes TypeScript compiler to fail and block users. With this PR I want to introduce some tests to prevent merging of broken typings files.

## Types of changes

- [x] Polish
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

TODOs
- [ ] fix typings validation for jasmine
- [ ] fix typings validation for selenium-standalone-service
- [ ] raise ticket for execute and call typings in async mode

### CODE REVIEW

**Comment by mgrybyk** [comment]
@CrispusDH may you take a look?

File: scripts/type-generation/test/copy.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    '@types/selenium-standalone': 'packages/wdio-selenium-standalone-service/node_modules/@types/selenium-standalone',
+    '@wdio/selenium-standalone-service': 'packages/wdio-selenium-standalone-service',
+}
+
+/**
+ * copy package.json and typings from package to type-generation/test/.../node_modules
+ */
+async function copy() {
+    outDirs.forEach(async outDir => {
```
Comment (CrispusDH) [inline scripts/type-generation/test/copy.js]: You have to use either `for ... of` or `await Promise.all(packages.map())` construction. I prefer first option. `forEach` method does not in a proper way with async ... await https://stackoverflow.com/questions/37576685/using-async-await-with-a-foreach-loop

Comment (christian-bromann) [reply scripts/type-generation/test/copy.js]: keeping this open until this is resolved.

Comment (mgrybyk) [reply scripts/type-generation/test/copy.js]: my bad, fixed

**Comment by mgrybyk** [comment]
working on TODOs right now

**Comment by christian-bromann** [comment]
@mgrybyk please remove the wip label when ready to review

**Comment by CrispusDH** [comment]
I think it might be useful to create a short README.md about "How we check types". Some steps or ideas. It would be helpful for contributors if future. @mgrybyk , what do you think? I mean it could be difficult to figure out what commands are executes and what happens in files.

**Comment by mgrybyk** [comment]
@CrispusDH added README.md. Is it helpful? What kind of information is it missing?

**Comment by christian-bromann** [comment]
@mgrybyk actually would you mind to move the type generation tests to `/tests/`? I think having smoke and type checks in one place is better than having them across the project.

**Comment by mgrybyk** [comment]
I'll take a look in next couple of days or even earlier

**Pull request merged**